force-app/main/default/lwc/datatableInlineEditWithUiApi/datatableInlineEditWithUiApi.js

```diff
@@ -0,0 +1,66 @@
+import { LightningElement, wire } from 'lwc';
+import getContacts from '@salesforce/apex/ContactController.getContactList';
+import { refreshApex } from '@salesforce/apex';
+import { updateRecord } from 'lightning/uiRecordApi';
+
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+import ID_FIELD from '@salesforce/schema/Contact.Id';
+import FIRSTNAME_FIELD from '@salesforce/schema/Contact.FirstName';
+import LASTNAME_FIELD from '@salesforce/schema/Contact.LastName';
+import TITLE_FIELD from '@salesforce/schema/Contact.Title';
+import PHONE_FIELD from '@salesforce/schema/Contact.Phone';
+import EMAIL_FIELD from '@salesforce/schema/Contact.Email';
+
+const COLS = [
+    { label: 'First Name', fieldName: 'FirstName', editable: true },
+    { label: 'Last Name', fieldName: 'LastName', editable: true },
+    { label: 'Title', fieldName: 'Title', editable: true },
+    { label: 'Phone', fieldName: 'Phone', type: 'phone', editable: true },
+    { label: 'Email', fieldName: 'Email', type: 'email', editable: true }
+];
+export default class DatatableInlineEditWithUiApi extends LightningElement {
+    columns = COLS;
+    draftValues = [];
+
+    @wire(getContacts)
+    contacts;
+
+    handleSave(event) {
+        const fields = {};
+        fields[ID_FIELD.fieldApiName] = event.detail.draftValues[0].Id;
```

Comment:
Here, the updateRecord will be called multiple times in its own transaction. The UI API might be the best option for a single record update. Apex with a single DML might be better for multiple records updates. Shall I go ahead with this change?

Reply:
I agree that having separate transactions is not ideal but is there a way to limit the inline editing to a single row?
We can't have a UI behavior that doesn't match with the backend operation.

If we can't limit editing to a single row, I suggest adding a comment about the parallel transactions. This reinforces the difference with the Apex recipe.

Reply:
I find it odd this is the UIAPI recipe when it still uses Apex to load the list of records. I recognize that [`lightning/uiListApi`](https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.reference_lightning_ui_api_list_ui) is deprecated. Until a new wire adapter to fetch list view data I wonder if this UIAPI recipe should be removed.

Reply:
There is already a GA's successor to `lightning/uiListApi` => https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.reference_get_list_info_by_name

Reply:
Unfortunately and unlike its predecessor, `getListInfoByName` from `lightning/uiListsApi` is not so useful because it doesn't return the list of records.

For example here's what it returns:
```json
{
  "cloneable": true,
  "createable": true,
  "deletable": true,
  "displayColumns": [
    {
      "fieldApiName": "Name",
      "label": "Account Name",
      "sortable": true
    },
    {
      "fieldApiName": "Site",
      "label": "Account Site",
      "sortable": true
    },
    {
      "fieldApiName": "BillingState",
      "label": "Billing State/Province",
      "sortable": true
    },
    {
      "fieldApiName": "Phone",
      "label": "Phone",
      "sortable": true
    },
    {
      "fieldApiName": "Type",
      "label": "Type",
      "sortable": true
    },
    {
      "fieldApiName": "Owner.Alias",
      "label": "Account Owner Alias",
      "sortable": true
    }
  ],
  "filterLogicString": null,
  "filteredByInfo": [],
  "label": "All Accounts",
  "listReference": {
    "id": "00B9A000004m73MUAQ",
    "listViewApiName": "AllAccounts",
    "objectApiName": "Account",
    "type": "listView"
  },
  "orderedByInfo": [
    {
      "fieldApiName": "Name",
      "isAscending": true,
      "label": "Account Name"
    }
  ],
  "updateable": true,
  "userPreferences": {
    "columnWidths": {
      "Site": -1,
      "Type": -1,
      "Owner.Alias": -1,
      "Phone": -1,
      "BillingState": -1,
      "Name": -1
    },
    "columnWrap": {
      "Site": false,
      "Type": false,
      "Owner.Alias": false,
      "Phone": false,
      "BillingState": false,
      "Name": false
    }
  },
  "visibility": "Public",
  "visibilityEditable": true
}
```